### PR TITLE
fix(connectors): fixes MGDCTRS-1129, MGDCTRS-1128: cleaned up search and order by fields, added support for ILIKE comparision

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Cleaned up search and order by fields for cluster, namespaces, types and connectors
Added support for ILIKE comparision operator in search parameter

## Verification Steps
CI tests should pass.

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
~~- [ ] Verified independently by reviewer~~
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: pkg/services/queryparser/query_parser.go [hunk 1]
```diff
@@ -74,6 +76,7 @@ var _ QueryParser = &queryParser{}
 // EQ           -> VALUE | QUOTED_VALUE
 // NOT_EQ       -> VALUE | QUOTED_VALUE
 // LIKE         -> VALUE | QUOTED_VALUE
+// ILIKE        -> VALUE | QUOTED_VALUE
```
Comment (ziccardi) [inline pkg/services/queryparser/query_parser.go:79]: Please update https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1203/files#diff-b9f3832eada41015efaf6287638235412d9395abab3f46d884919979457dcdebR75 too to:
```
// COLUMN       -> EQ | NOT_EQ | LIKE | ILIKE
```

**Comment by ziccardi** [comment]
@machi1990 @pawelpaszki This PR is adding the `ILIKE` SQL keyword to the `QueryParser` and thus to the Kafka list endpoint as well. I don't think it is an issue, but I'd like to have your opinion.

**Comment by pawelpaszki** [comment]
> @machi1990 @pawelpaszki This PR is adding the `ILIKE` SQL keyword to the `QueryParser` and thus to the Kafka list endpoint as well. I don't think it is an issue, but I'd like to have your opinion.

looks good. thanks for adding it to the openapi spec as well

**Comment by machi1990** [comment]
> @machi1990 @pawelpaszki This PR is adding the `ILIKE` SQL keyword to the `QueryParser` and thus to the Kafka list endpoint as well. I don't think it is an issue, but I'd like to have your opinion.

Thanks @ziccardi, can we add an integration test in the for the Kafka codebase as well?

File: pkg/services/queryparser/query_parser.go (hunk 1 above)
Comment (dhirajsb) [reply pkg/services/queryparser/query_parser.go:79]: Good catch. Fixed.

**Review by valdar — CHANGES_REQUESTED** [review]
Looks good to me. 
I would just add some tests to check the additions, at least where the text query field name substitution is involved.

**Review by valdar — APPROVED** [review]

**Pull request merged**